files/Aboot/boot0.j2

```diff
@@ -317,106 +444,150 @@ platform_specific() {
+        cmdline_add "hwaddr_ma1=$(cat /sys/class/net/eth0/address)"
     else
         err "Management port not found."
     fi
 
-    # use extra parameters from kernel-params hook if the file exists
-    if [ -f "$target_path/$kernel_params" ]; then
-        cat "$target_path/$kernel_params" >> /tmp/append
+    # Obtain root partition uuid
+    local rootdev="$(mount | grep -E "$flash_re" | cut -f1 -d' ')"
+    local rootfstype="$(mount | grep -E "$flash_re" | cut -f5 -d' ')"
+    local rootuuid="$(get_uuid_for $rootdev)"
+    if [ -z "$rootuuid" ] || [ "$rootfstype" = "vfat" ] ; then
+        cmdline_add "root=$rootdev"
+    else
+        cmdline_add "root=UUID=$rootuuid"
     fi
+}
 
-    # setting root partition if not overridden by kernel-params
-    if ! grep -q "root=" /tmp/append; then
-        rootdev="$(mount | grep -E "$flash_re" | cut -f1 -d' ')"
-        rootfstype="$(mount | grep -E "$flash_re" | cut -f5 -d' ')"
-        rootuuid="$(get_uuid_for $rootdev)"
-        if [ -z "$rootuuid" ] || [ "$rootfstype" = "vfat" ] ; then
-            echo "root=$rootdev" >> /tmp/append
-        else
-            echo "root=UUID=$rootuuid" >> /tmp/append
-        fi
+write_default_cmdline() {
+    local delimiter="cmdline-aboot-end"
+
+    cmdline_clear
+
+    if $in_aboot; then
+        # generate the default kernel parameters for the platform
+        cat /etc/cmdline | sed "/^\(${bootconfigvars// /\|}\|crashkernel\|loglevel\|ignore_loglevel\)\(\$\|=\)/d;/^\$/d" | cmdline_append
+    elif grep -q "$delimiter" /proc/cmdline; then
+        # we are on a recent sonic image using delimiter. extracting the part of the
+        # cmdline coming from aboot is trivial.
+        cat /proc/cmdline | sed -E "s/^(.*) $delimiter .*$/\1/" | tr ' ' '\n' | cmdline_append
+    else
+        # we are either on SONiC or EOS and the commandline doesn't have a delimiter
+        # for the Aboot part. Take an educated guess at a right delimiter.
+        # Subject to breakage if EOS or SONiC cmdline change.
+        cat /proc/cmdline | sed -E 's/^(.*) rw .*$/\1/' | tr ' ' '\n' | cmdline_append
     fi
 
-    mkdir -p "$image_path"
-    cat /tmp/append > $cmdline_image
-    [ -s ${target_path}/machine.conf ] || write_machine_config
-    sync
+    cmdline_add "$delimiter"
+}
+
+write_common_configs() {
+    write_default_cmdline
+    write_platform_specific_cmdline
+    write_image_specific_cmdline
+    write_machine_config
+}
+
+write_secureboot_configs() {
+    write_common_configs
+    cmdline_add "loop=${installer_image_path/mnt\/flash\//}"
+    cmdline_add loopfstype=squashfs
+    cmdline_add "loopoffset=$(unzip -qqf "$installer_image_path" fs.squashfs)"
+    cmdline_add docker_inram=on
+    cmdline_add secureboot=on
+    cmdline_add aboot.secureboot=enabled
+    # reboot after 60 seconds when a panic occurs. this has the desired side effect
+    # of disabling the interactive shell in the initramfs
+    cmdline_add panic=60
```

Comment:
What the user could attack in the 60 second? If nothing to attack, what are the PRO/CON for longer/shorter timespan?

Reply:
Nothing, the initramfs just gets stuck there with the error message and reboots after 60 seconds.
I'm not entirely sure about pro/cons, I believe some tasks might need some time to cleanup.

The default behavior in SONiC is to wait indefinitely `panic=0`, the default is configured by the sonic-linux-kernel configuration `CONFIG_PANIC_TIMEOUT=0` (see `kernel/panic.c`)

I can revert it to the default behavior of `0` if you think it's better.
As long as the parameter is defined the cmdline, the initramfs will lock the console access.

Reply:
The default behavior is more sensible. Otherwise the continuously rebooting switch seems broken.